### DISCUSSION TITLE
NuGet does not find LangChain.Chains.Chain

### DISCUSSION THREAD

**Discussion by belenmatto** [tryAGI/LangChain#172] [Q&A]
Hi! How are you?

I'm trying to implement a chatbot using Langchain on C#. I imported all packages but I get a problem with LangChain.Chains.Chain. This does not appear into NuGet packages admin, inclusive if I keep the active selection "Include preliminar versions".
I am working with NuGet 5.11.4 and .NET 4.8.

Is it a version problem, maybe? Has happened to them previously?
Thank you!

![image](https://github.com/tryAGI/LangChain/assets/133797166/3aa78629-552c-4c2b-aa4d-e03660ab1101)

**Comment by curlyfro** [comment]
i haven't tried with .NET Framework but try adding to your using statements:

using static LangChain.Chains.Chain;

**Reply by belenmatto** [reply]
I have version problems with PdfPig too because is not supported in  .NET ^4.6, and LangChain depends directly of it. Thanks for your time!

**Comment by HavenDV** [comment]
> I have version problems with PdfPig too because is not supported in .NET ^4.6, and LangChain depends directly of it. Thanks for your time!

Hello @belenmatto. PdfPig is delivered by LangChain.Sources.Pdf and is not included in the standard LangChain package.
That package also has AsposePdfSource which you can try and it may work for you.
Feedback is important to us, we have already encountered problems in PdfPig and we are interested in a stable implementation of PdfSource, so please let us know about it
